templates/azure/mender-artifact-upload.yml

```diff
@@ -0,0 +1,59 @@
+# Use this template to upload a Mender Artifact to a Mender server.
+#
+# The template is Azure DevOps step template with parameters. It's designed to be included
+# from the GitHub repository.
+#
+# The template has the following parameters:
+# - `mender_uri`:      Mender server URL.
+# - `mender_pat`:      Mender server access token.
+# - `mender_artifact`: Relative to $(System.DefaultWorkingDirectory) path of Mender Artifact file
+
+parameters:
+- name: 'mender_uri'
+  type: string
+  default: 'https://hosted.mender.io'
+- name: 'mender_pat'
+  type: string
+- name: 'mender_artifact'
+  type: string
+
+steps:
+- script: |
```

Comment:
It is all one single step, right? Then I think in the README we should not insist in that they are steps based. For example "Azure DevOps step templates with parameters" or "The step templates gives". They could just be "DevOps templates".

Otherwise we can split it in several steps, which might be the recommended practice? Not clear to me from the documentation linked above :shrug:

Reply:
Actually this `step templates` in docs is intentional because in Azure DevOps is possible to make templates for steps, jobs and stages ([docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#step-reuse)). It might be that it doesn't sound habitually for us, but I believe for Azure DevOps community it will be meaningful.